### ISSUE TITLE
vCard Export Telefon-Nummern

### ISSUE THREAD

**Issue by Michael-Schaer** [hitobito/hitobito#710]
Beim Personen-Export (vCard) werden im Moment keine Tel.-Nummern exportiert. Wenn ich die Berechtigung habe, um eine Telefonnummer zu sehen, sollte diese m.E. auch im Export drin sein. Siehe auch Vergleich Screenshot.

Dazu kommt ein Problem bei den Umlauten. Dies ist womöglich aber Outlook-Seitig.

![vcard](https://user-images.githubusercontent.com/1799621/51685065-d30c0000-1fed-11e9-8d6e-c5d8a589e9b9.png)

**Comment by RolandStuder** [comment]
Ich kann das Problem nicht reproduzieren, bei meinen Tests sind die Telefonnummern im .vcf file drin, kannst du mir das Beispiel per e-Mail schicken?

Ich konnte allerdings nur auf OS X testen. Möglicherweise ein allgemeines Encoding Problem.

**Comment by Michael-Schaer** [comment]
Ich glaube das Problem liegt bei der Sprache. Im Export wird der Tel-Type in Deutsch angegeben:

> TEL;TYPE=arbeit:031 328 05 51

Bei vCard 3.0 sollten die Typen ja vordefiniert sein:

```
tel-type     = "HOME" / "WORK" / "PREF" / "VOICE" / "FAX" / "MSG"
                / "CELL" / "PAGER" / "BBS" / "MODEM" / "CAR" / "ISDN"
                / "VIDEO" / "PCS" / iana-token / x-name
```

Die meisten Applikationen erwarten wohl "home" oder "work". Da wir in der MiData fixe Telefon-Typen haben, könnten wir diese beim Export doch in einen gängigen Type matchen, nicht?

![tel-types](https://user-images.githubusercontent.com/1799621/51738400-63e6e800-208f-11e9-88f3-bc104420c8d9.png)

**Comment by RolandStuder** [comment]
@Michael-Schaer Ja, ich sehe. Wir werden für die toleranteren Programme die eigenen Telefontypen verlieren, aber wenn es für gewisse Programme gar nicht funktioniert, ist das natürlich doof. Kannst du lokal das .vcf File anpassen und schauen, ob es mit den anderen Typen korrekt importiert?

**Comment by Michael-Schaer** [comment]
Hallo Roland, folgende vCard funktioniert problemlos in Outlook Version 1812:

```
BEGIN:VCARD
VERSION:3.0
N;CHARSET=utf-8:Schär;Michael;;;
FN;CHARSET=utf-8:Michael Schär
BDAY:19901018
ADR;CHARSET=utf-8:;;Spöüästrasse 12é;Bern;;3011;
EMAIL;TYPE=pref:example@mail.com
TEL;TYPE=work:031 328 05 51
END:VCARD
```

Geändert habe ich folgendes:
- Tel type -> "work"
- CHARSET=utf-8 bei Namen und Adresse

**Comment by js4gh** [comment]
Umlaute werden nicht richtig interpretiert und Telefonnummern fehlen
Windows 10 Pro Outlook 365

Datei wird scheinbar falsch generiert.
https://tools.ietf.org/html/rfc6350#section-6.4.1

@carlobeltrame kannst du das als "Bug" aufnehmen?

**Comment by cdn64** [comment]
Leider scheint es nicht möglich sein, das `charset` bei der Adresse mitzugeben: https://github.com/qoobaa/vcard/blob/master/lib/vcard/vcard.rb#L124. Zudem ist die Library / das Gem nicht mehr maintained… Das heisst, dass wir das entweder forken und ergänzen, oder die ganze Funktionalität anders / selber abdecken müssten.

Der `type` der Telefonnummer hingegen könnte schon einfach auf `work` gesetzt werden, oder auf ein anderes der «[predefined values](https://tools.ietf.org/html/rfc6350#page-35)», die wohl alle unter Outlook funktionieren würden. Dabei ginge aber die Information des Labels verloren, die bspw. auf macOS schön übernommen wird.

**Comment by ThomasEllenberger** [comment]
Da diese Diskussion bereits älter ist und von unserer Seite her aktuell kein Plan existiert dies umzusetzen, schliesse ich dieses Issue vorerst, um die Anzahl offener Issues möglichst übersichtlich zu halten.
Kann bei Bedarf wieder geöffnet werden.
